Code/src/main/java/nl/utwente/group10/ui/components/Connection.java

```diff
@@ -206,4 +215,23 @@ public String toString() {
         return "Connection connecting \n(out) " + startAnchor + "   to\n(in)  "
                 + endAnchor;
     }
+    
+    /**
+     * This method evaluates the validity of the created connection.
+     * If the connection results in an invalid operation a visual
+     * error will be displayed.
+     */
+    public void checkError() {
+        if(startAnchor.isPresent() && endAnchor.isPresent()) {
+            try {
+                //TODO Obviously this will cause errors, we need a way to access the Env
+                endAnchor.get().getBlock().asExpr().analyze(new Env(), new GenSet());
+                this.setStyle("-fx-stroke: black;");
+            } catch (HaskellTypeError e) {
+                this.setStyle("-fx-stroke: red;");
+            } catch (HaskellException e) {
+                //e.printStackTrace();
```

Comment:
Ik weet niet precies wat je bedoelt, maar ik bedoelde met deze comment specifiek de `//e.printStackTrace`.

Als je een HaskellTypeException uit analyze krijgt heeft de gebruiker iets verkeerd gedaan en moet die dus zo goed mogelijk en op een nette manier op de hoogte gesteld worden. Ik kan me voorstellen dat dat complex is.

Als je een HaskellException uit analyze krijgt hebben *wij* iets verkeerd gedaan en moeten dus alle alarmlichten afgaan. Wat mij betreft mag je stack traces printen, labels met foutmeldingen op het scherm toveren, je eigen `alert()` bouwen, wat je maar wilt.

Dat gebeurt nu nog niet echt.